simulation.py

```diff
@@ -0,0 +1,84 @@
+"""
+Fuctions for a process to simulate an alpha.
+"""
+
+import copy
+import importlib
+import sys
+from thousandaire.constants import DATA_LIST_ALL
+from thousandaire.constants import TRADING_CONFIG, TRADING_REGION
+from thousandaire.data_loader import DataLoader
+from thousandaire.simulator import Simulator
+
+PRICE_DATASET = 'price_dataset'
+
+def initialize():
+    """
+    Initialize data we get.
+    """
+    raw_data = DataLoader(DATA_LIST_ALL).get_all()
+    workdays = {
+        region : raw_data['workdays'][region]
+        for region in TRADING_REGION if region in raw_data['workdays'].keys()}
+    return_data = {}
+    for key, value in workdays.items():
+        return_data[key] = copy.deepcopy(raw_data)
+        for dataset in return_data[key].values():
+            dataset.set_workdays(value)
+    return_data['workdays'] = workdays
+    return return_data
+
+def pure_data(raw_data, data_list, target, region):
+    """
+    Pure the data which alpha need.
+
+    Due to the raw_data is combined with all data we have, we need to pure
+    the data ,which the alpha need, to pass for saving resource.
+    """
+    workdays = raw_data['workdays'][region]
+    if target in raw_data[region].keys():
+        target_data = raw_data[region][target]
+    else:
+        raise KeyError("Key not found")
+    simulate_data = {
+        name : raw_data[region][name]
+        for name in data_list if name in raw_data[region].keys()}
+    data = {
+        'workdays' : workdays,
+        'target_data' : target_data,
+        'simulate_data' : simulate_data}
+    return data
+
+def simulation(alpha_path, raw_data):
+    """
+    Act the process of simulating.
+    """
+    try:
+        setting_module = importlib.import_module(alpha_path).AlphaSettings()
+    except FileNotFoundError:
+        print("Alpha did not exist.")
+    data = pure_data(
+        raw_data,
+        setting_module.data_list,
+        TRADING_CONFIG[(setting_module.target[0],
+                        setting_module.target[1])][PRICE_DATASET],
+        setting_module.target[1])
+    simulator = Simulator(setting_module, data)
+    results = simulator.run()
+    return results
+
+def main():
+    """
+    Run the simulation process.
+    """
+    args = sys.argv
+    target = 'settings'
+    for item in args:
+        if item.find(target) != -1:
+            alpha_path = item
+    data = initialize()
```

Comment:
(In the current version,) initialization is needed only when we want to simulate something. So it's better to put this line in simulate().